modules/enterprise/server/jar/src/main/java/org/rhq/enterprise/server/configuration/ConfigurationManagerBean.java

```diff
@@ -1568,7 +1570,9 @@ public void completeResourceConfigurationUpdate(ConfigurationUpdateResponse resp
         } else if (response.getStatus() == ConfigurationUpdateStatus.SUCCESS) {
             // link to the newer, persisted configuration object
             Resource resource = update.getResource();
+            Configuration oldResourceConfig = resource.getResourceConfiguration();
             resource.setResourceConfiguration(update.getConfiguration().deepCopyWithoutProxies());
+            entityManager.remove(oldResourceConfig);
```

Comment:
Same question/concern as before. Wouldn't updated resource configuration simply result in the previous configuration history to appear on the resource's configuration history page?